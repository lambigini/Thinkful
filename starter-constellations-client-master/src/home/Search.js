import React from "react";

function Search ({ constellations, setConstellations }) {
  const filterConstellations = (event) => {
    const searchString = event.target.value;
    const visible = constellations.all.filter(({ name }) =>
      name.toLowerCase().includes(searchString)
    );

    setConstellations({
      ...constellations,
      visible,
    });
  };

  return (
    <form className="form container">
      <div className="form-group row">
        <input
          className="form-control col col-md-8"
          type="text"
          id="search"
          name="search"
          onChange={filterConstellations}
          placeholder="Search the stars..."
        />
        <label className="form-check-label sr-only" htmlFor="search">
          Search
        </label>
      </div>
    </form>
  );
};

export default Search;
