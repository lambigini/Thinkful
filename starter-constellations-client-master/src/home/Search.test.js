import React from "react";
import Search from "./Search";
import renderer from "react-test-renderer";

it("renders the Search", () => {
  const component = renderer.create(<Search />);
  expect(component.toJSON()).toMatchSnapshot();
});
