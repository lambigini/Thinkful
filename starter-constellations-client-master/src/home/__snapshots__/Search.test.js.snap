// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Search 1`] = `
<form
  className="form container"
>
  <div
    className="form-group row"
  >
    <input
      className="form-control col col-md-8"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Search the stars..."
      type="text"
    />
    <label
      className="form-check-label sr-only"
      htmlFor="search"
    >
      Search
    </label>
  </div>
</form>
`;
